File: android/SeatCanvasSample/libseatcanvas/src/main/java/com/libseatcanvas/style/CircleSeatStyleConfig.kt
package com.libseatcanvas.style

import androidx.annotation.ColorInt

/**
 * 圆形座位样式。选中态可使用 [overlay]、[checkmark]；未选中一般只需 [fill]。
 * @param fill 填充色（ARGB）
 * @param overlay 遮罩色，选中时覆盖在座位上；未使用则为 null
 * @param checkmark 对勾颜色；未使用则为 null
 */
data class CircleSeatStyleConfig(
    @ColorInt val fill: Int,
    @ColorInt val overlay: Int? = null,
    @ColorInt val checkmark: Int? = null
) : SeatStyleConfig {
    override val type: SeatStyleType = SeatStyleType.CIRCLE
}

File: src/SeatCanvas/platform/apple/swift/CircleSeatStyleConfig.swift
//
//  CircleSeatStyleConfig.swift
//  SeatCanvas
//
//  Created by king on 2026/1/15.
//

import Foundation

struct CircleSeatStyleConfig: SeatStyleConfig {
    let type: SeatStyleType = .circle
    let fill: PlatformColor
    let overlay: PlatformColor?
    let checkmark: PlatformColor?

    init(fill: PlatformColor, overlay: PlatformColor? = nil, checkmark: PlatformColor? = nil) {
        self.fill = fill
        self.overlay = overlay
        self.checkmark = checkmark
    }

    enum CodingKeys: String, CodingKey {
        case type
        case fill
        case overlay
        case checkmark
    }
}

extension CircleSeatStyleConfig: Encodable {
    func encode(to encoder: Encoder) throws {
        var container = encoder.container(keyedBy: CodingKeys.self)
        try container.encode(type, forKey: .type)
        try container.encode(fill.argbHex, forKey: .fill)
        try container.encodeIfPresent(overlay?.argbHex, forKey: .overlay)
        try container.encodeIfPresent(checkmark?.argbHex, forKey: .checkmark)
    }
}
